assets/scripts/app/PrintContainer.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { attachPrintEventListeners } from './print'
+import { getStreetImage } from '../streets/image'
+
+class PrintContainer extends React.PureComponent {
+  constructor (props) {
+    super(props)
+
+    this.createPrintImage = this.createPrintImage.bind(this)
+  }
+
+  componentDidMount () {
+    attachPrintEventListeners()
```

Comment:
I actually did have the listeners moved into this component (https://github.com/streetmix/streetmix/pull/696/commits/82e2c91a499f901efd4a4d5a2ef30118417bab21) but moved it back out to avoid having to having to bind any action creators and store to this component at all, since the module it imports from is already doing it.

It's not a big deal to revert this if you think it's better to include it here anyway.

As for arrow functions to avoid `.bind(this)` in the constructor, that's interesting. It looks like this syntax is part of the [Class Fields](https://github.com/tc39/proposal-class-fields) proposal, currently at [stage 3](https://github.com/tc39/proposals). I think this is the same proposal that would allow us to declare `propTypes` and `defaultProps` as static properties on the class (rather than attaching as object properties after the class is instantiated), something that I'd been considering separately as well -- but we currently don't adopt the Babel transform that permits this.

I think this might be out of scope for this PR, but I think we could certainly consider adopting the transform in a separate one?